codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -833,8 +834,18 @@ public Http2Stream closeRemoteSide() {
      */
     private final class DefaultEndpoint<F extends Http2FlowController> implements Endpoint<F> {
         private final boolean server;
+        /**
+         * This is the next stream id that this endpoint expects to see when creating a new stream.
+         */
         private int nextStreamId;
-        private int lastStreamCreated;
+        /**
+         * Used to generate stream ids. This is different from {@link #nextStreamId} because this may be used to
+         * create stream ids for stream that may not be created immediately. For example if events need to be queued
+         * to respect the
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.5.2">SETTINGS_MAX_CONCURRENT_STREAMS</a> this can be
+         * used to ensure stream ids continue to increase.
+         */
+        private int nextStreamIdToCreate;
```

Comment:
Maybe something like:

```java
/**
    The minimum stream ID allowed when creating the next stream. This only
    applies at the time the stream is created. If the ID of the stream being
    created is less than this value, stream creation will fail. Upon
    successful creation of a stream, this value is incremented to the next
    valid stream ID.
 */
```

Also I think the field would read better as `nextStreamToCreate`.  Thoughts?